moped-database/migrations/1644020502599_backfill_moped_proj_features_fk/up.sql

```diff
@@ -0,0 +1,9 @@
+-- populate proj_feature <> component fk (project_component_id) from previous
+-- associative table (moped_proj_features_components)
+update moped_proj_features
+	set project_component_id = moped_proj_features_components.moped_proj_component_id
+	from moped_proj_features_components
+	where moped_proj_features_components.moped_proj_features_id = moped_proj_features.feature_id;
+
+-- and now it's safe to require project_component_id for all features
+ALTER TABLE "public"."moped_proj_features" ALTER COLUMN "project_component_id" SET NOT NULL;
```

Comment:
> ALTER TABLE "public"."moped_proj_features" ALTER COLUMN "project_component_id" SET NOT NULL;

because our mapping schema has evolved significantly since day one, i'm not 100% sure this constraint will succeed in staging. i could split it into a separate migration in case we want to exclude it? i could also run the migration against a dump of staging. i should do at least one of these things before merging i think.